src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/TableReadAttemptBuilder.java

```diff
@@ -61,13 +61,27 @@ public ReadAttempt<TableSchemaObject> build(WhereCQLClause<Select> whereCQLClaus
 
     readPosition += 1;
 
-    return new ReadAttempt<>(
-        readPosition,
-        tableSchemaObject,
-        selectCQLClause,
-        whereCQLClause,
-        cqlOptions,
-        pagingState,
-        documentSourceSupplier);
+    final WhereCQLClauseAnalyzer.WhereCQLClauseAnalyzeResult whereCQLClauseAnalyzeResult =
+        whereCQLClause.analyseWhereClause();
+    // Analyse filter usage
+    // Step 1, may add AllowFiltering
+    if (whereCQLClauseAnalyzeResult.withAllowFiltering()) {
+      cqlOptions.addBuilderOption(CQLOption.ForSelect.withAllowFiltering());
+    }
+
+    var tableReadAttempt =
+        new ReadAttempt<>(
+            readPosition,
+            tableSchemaObject,
+            selectCQLClause,
+            whereCQLClause,
+            cqlOptions,
+            pagingState,
+            documentSourceSupplier);
+
+    // Step 2, may add warning
+    whereCQLClauseAnalyzeResult.warnings().forEach(tableReadAttempt::addWarning);
```

Comment:
not sure on this, certainly the builder is doing the right thing doing the analysis, just nto sure about adding the warnings like this. Nothingt to change for now, just commenting from our discussion